lib/plugins.js

```diff
@@ -22,19 +22,24 @@ var log_ = function() {
  * @param {Object} config parsed from the config file
  */
 var Plugins = function(config) {
+  var self = this;
+
   this.pluginConfs = config.plugins;
-  this.pluginObjs = {};
-  for (var name in this.pluginConfs) {
-    var pluginConf = this.pluginConfs[name];
+  this.pluginObjs = [];
+  this.pluginConfs.forEach(function(pluginConf) {
     var path;
     if (pluginConf.path) {
       path = ConfigParser.resolveFilePatterns(pluginConf.path, true,
           config.configDir)[0];
     } else {
-      path = pluginConf.package || name;
+      path = pluginConf.package;
+    }
+    if (!path) {
+      throw new Error('Plugin configuration did not contain a valid path.');
     }
-    this.pluginObjs[name] = require(path);
-  }
+    pluginConf.name = path;
```

Comment:
The file name is usually just 'index.js', which isn't very helpful.

We could use some heuristic to try to trim the last folder + file name so that 'foo/bar/baz/index.js' turns into 'baz/index.js'. Does that seem better?